src/nvim/statusline.c

```diff
@@ -1554,6 +1573,34 @@ int build_stl_str_hl(win_T *wp, char *out, size_t outlen, char *fmt, char *opt_n
       }
       break;
 
+    case STL_FOLDCOL:    // 'C' for 'statuscolumn'
+    case STL_SIGNCOL: {  // 's' for 'statuscolumn'
+      if (stcp == NULL) {
+        break;
+      }
+
+      bool fold = opt == STL_FOLDCOL;
+      *buf_tmp = NUL;
+      for (int i = 0; i <= 9; i++) {
+        char *p = fold ? stcp->fold_text : stcp->sign_text[i];
+        if (*p == NUL && *buf_tmp == NUL) {
+          break;
+        }
+        stl_items[curitem].type = Highlight;
+        stl_items[curitem].start = out_p + STRLEN(buf_tmp);
+        stl_items[curitem].minwid = *p == NUL ? 0 : fold ? stcp->fold_attr
```

Comment:
You should store the signcolumn width as that needs to remain constant across all lines.

Reply:
Am setting empty sign cells in the signcolumn to `"  "` [instead](https://github.com/neovim/neovim/compare/7b4ed497c591b240df492a73c2b09d79770d5ace..dc855059f3af8b7b6a95d2d5023a3273a147df79#diff-f8a0f2dc4a1b93961afa1ab450afc474ca5d1fbfb41d074a77715d19c8f492b1R430-R432).